src/Checks.js

```diff
@@ -90,18 +87,24 @@ class CheckForm extends Component {
     }
 
     fields.push(<SingleFieldElement 
-          label={label}
-          value={value}
-          controlId={controlId}
-          onChange={onChange}
-          placeholder={placeholder}
-          key='control'
-        />);
+                    label={label}
+                    value={value}
+                    controlId={controlId}
+                    onChange={onChange}
+                    placeholder={placeholder}
+                    key='control'
+                />);
 
     return (
       <ResourceForm data={this.state} baseResource={this.props.baseResource}>
+        <h2>New Check</h2>
+        <FormGroup controlID="checkName">
+          <ControlLabel>Name</ControlLabel>
+          <FormControl value={this.state.check_name} placeholder="Enter a name for your check" />
```

Comment:
Question - how does this in turn change the check_name on this.state without an onChange? and why not just use the SingleFieldElement wrapper?